doc/vlan/switchport-mode-support/Switchport Mode and VLAN CLI Enhancement.md

```diff
@@ -0,0 +1,608 @@
+
+This high-level design document describes the implementation of Switchport modes and VLAN CLI enhancements in SONiC.
+
+## Definitions/Abbreviations
+
+| **Sr No** |  **Term**   |  **Definition**                      |
+| :------------- |:-------------| :-----|
+|  1       |  VLAN_ID      |  Unique identifier for each VLAN     |  
+|  2       |  Trunk  |  The port is untagged member to one VLAN and tagged member to one or more VLANs  |  
+|  3       | Access |  The port is untagged member to only one VLAN|  
+|  4       |  Routed | The port is in L3 interface mode                       |  
+
+
+
+## Overview
+
+This HLD will provide implementation details of switch port modes access and trunk on a Port or on a PortChannel. In this HLD vlan configuration commands are also improved for efficient configuration and management of VLANs.
+
+## Introduction
+
+Switch ports are Layer 2-only interfaces associated with a physical port. A switch port can have three modes: access port, a trunk port, or a hybrid port. We can manually configure a port as an access port, trunk port or hybrid port depending on requirement . Switch ports are used for managing the physical interface and associated Layer 2 protocols and do not handle routing.
+
+### Access Mode
+
+In access mode a port possesses the following properties
+* Receives untagged traffic for a particular Vlan
+* Forwards untagged traffic for a particular Vlan
+
+### Trunk Mode
+
+In trunk mode a port possesses the following properties
+* Receives untagged traffic for a single Vlan
+* Forwards untagged traffic for a single Vlan
+* Receives tagged traffic for one or more Vlan(s)
+* Forwards tagged traffic for one or more Vlan(s)
+
+### Sample Topology
+
+![Topology](https://user-images.githubusercontent.com/61490193/208155234-3a04ca08-5ca3-42b3-b679-c1efcacddff2.png)
+
+
+__*Figure 1: Sample Topology*__
+
+The figure shows that 3 different VLANs are configured with 6 other hosts. The switch ports connected to hosts would be configured as access ports. The ports that connect switches together would be configured as trunk ports.
+
+## Use Caess for Switchport modes
+### Access Mode
+Here are some common use cases for switch port mode access:
+
+**1.** Connecting devices to a network: When users need to connect devices to a switch, such as computers, printers, or IP phones, you can configure the switch ports to operate in access mode. This allows the devices to communicate with other devices on the network.
+
+**2.** Restricting VLANs: Users can use switch port mode access to restrict the VLANs that a device can access. This can help control access to network resources and improve network security.
+
+**3.** Improving network performance: By using switch port mode access, users can reduce the amount of unnecessary broadcast traffic on the network. When a switch port is in access mode, it only sends traffic to the device connected to that port, rather than broadcasting traffic to all devices on the network.
```

Comment:
What was stated was not correct... it only reduced flooding to the VLAN that this port belongs to which is one single VLAN.  if you had all devices connected to this same VLAN, it does not really reduce the flooding traffic.  you may want to re-format this...

Reply:
Ok, will re-format it